core/integration/bookkeeping/plugin.go

```diff
@@ -344,8 +344,7 @@ func (p *Plugin) CallStack(data interface{}) (stack map[string]interface{}) {
 			return
 		} else {
 			var updatedRun string
-			emptyTime := time.Time{}
-			if timeO2Start == emptyTime {
+			if function, ok := varStack["__call_func"]; ok && function == "UpdateRunStop" {
```

Comment:
I believe `function == "UpdateRunStop"` risks failing (I think the content of `__call_func` might include "`bookkeeping.`" as well) you should use `strings.Contains(function, "UpdateRunStop")` instead.